System eigen

A project which builds against ITK with system Eigen fails with `c:\misc\bluequartz\sdk-rwdi\itk-src\modules\core\common\include\itkSymmetricEigenAnalysis.h(22): fatal error C1083: Cannot open include file: 'itk_eigen.h': No such file or directory (compiling source file C:\Misc\BlueQuartz\DREAM3D_Plugins\ITKImageProcessing\ITKImageProcessingFilters\ITKImageReader.cpp)`. The ITK is also built [without default modules](https://github.com/dzenanz/DREAM3DSuperbuild/blob/f1f67e665c23833aabda27cc48a6e54a2ae13afd/projects/ITK.cmake#L90-L96).

Please reopen this
CMake Error at CMake/ITKModuleMacros.cmake:468 (add_library):
  Target "ITKCommon" links to target "Eigen3::Eigen" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  Modules/Core/Common/src/CMakeLists.txt:179 (itk_module_add_library)

[100%] Linking CXX executable ../../../../bin/itkTestDriver
/usr/bin/ld: cannot find -lEigen3::Eigen
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Modules/Core/TestKernel/src/CMakeFiles/itkTestDriver.dir/build.make:153: recipe for target 'bin/itkTestDriver' failed
make[2]: *** [bin/itkTestDriver] Error 1
CMakeFiles/Makefile2:15640: recipe for target 'Modules/Core/TestKernel/src/CMakeFiles/itkTestDriver.dir/all' failed
make[1]: *** [Modules/Core/TestKernel/src/CMakeFiles/itkTestDriver.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2

--- a/Modules/ThirdParty/Eigen3/itk-module-init.cmake
+++ b/Modules/ThirdParty/Eigen3/itk-module-init.cmake
@@ -1,0 +1,3 @@
+if(ITK_USE_SYSTEM_EIGEN)
+  find_package(Eigen3 REQUIRED CONFIG)
+endif()

